src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+
+    public OrderController(OrderService orderService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper,
+                           ShoppingCartService shoppingCartService,
+                           UserService userService) {
+        this.orderService = orderService;
+        this.orderResponseMapper = orderResponseMapper;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
+        Order order = orderService.completeOrder(shoppingCartService
```

Comment:
Consider avoid many nested method invocation (just a recommendation, not required).
It will make code readability and debugging easier.